src/main/java/com/scg/stop/auth/config/AuthUserArgumentResolver.java

```diff
@@ -0,0 +1,120 @@
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.support.WebDataBinderFactory;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.springframework.web.method.support.HandlerMethodArgumentResolver;
+import org.springframework.web.method.support.ModelAndViewContainer;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class AuthUserArgumentResolver implements HandlerMethodArgumentResolver {
+
+    private final UserRepository userRepository;
+    private final JwtUtil jwtUtil;
+
+    @Override
+    public Object resolveArgument(MethodParameter parameter, ModelAndViewContainer mavContainer,
+                                  NativeWebRequest webRequest, WebDataBinderFactory binderFactory)  {
+
+        HttpServletRequest request = webRequest.getNativeRequest(HttpServletRequest.class);
+
+        if (request == null) {
+            throw new InvalidJwtException(ExceptionCode.FAILED_TO_VALIDATE_TOKEN);
+        }
+        String contextPath = request.getContextPath();
+        String refreshToken = extractRefreshToken(request);
+        String accessToken = extractAccessToken(request);
+
+        //검증
+        if (jwtUtil.isAccessTokenValid(accessToken)) {
+            User extractedUser = extractUser(accessToken);
+            UserType extractedUserType = extractUser(accessToken).getUserType();
+
+            // TEMP USER 이고 REGISTER PATH 로 요청이면 REGISTER 페이지로 안내
+            // 그 외 TEMP USER 는 전부 예외 처리
+            String registerPath = "/register";
+            if (extractedUserType.equals(UserType.TEMP) ) {
+                if (contextPath.equals(registerPath)) {
+                    return extractedUser;
+                }
+                else {
+                    throw new BadRequestException(ExceptionCode.REGISTER_NOT_FINISHED);
+                }
+            }
+            AccessType[] allowedTypes = Objects.requireNonNull(parameter.getParameterAnnotation(AuthUser.class)).accessType();
+            List<AccessType> accessTypeList = Arrays.asList(allowedTypes);
+
+            if (accessTypeList.contains(AccessType.ALL)) {
+                return extractedUser;
+            }
+            else if (extractedUserType.equals(UserType.ADMIN)) {
+                if (accessTypeList.contains(AccessType.ADMIN)) return extractedUser;
+            }
+            else if (extractedUserType.equals(UserType.COMPANY) || extractedUserType.equals(UserType.INACTIVE_COMPANY)){
+                if (accessTypeList.contains(AccessType.COMPANY)) return extractedUser;
+            }
+            else if (extractedUserType.equals(UserType.PROFESSOR) || extractedUserType.equals(UserType.INACTIVE_PROFESSOR)) {
+                if (accessTypeList.contains(AccessType.PROFESSOR)) return extractedUser;
+            }
+            else if (extractedUserType.equals(UserType.STUDENT)) {
+                if (accessTypeList.contains(AccessType.STUDENT)) return extractedUser;
+            }
```

Comment:
회원 유형 중 "외부인"은 학생과 똑같은 권한을 같기로 해서 아래와 같이 해주면 될 것 같습니다

```suggestion
            else if (extractedUserType.equals(UserType.STUDENT) || extractedUserType.equals(UserType.OTHERS)) {
                if (accessTypeList.contains(AccessType.STUDENT)) return extractedUser;
            }
```

Reply:
893b4c8